src/app/agreement/agreement.component.ts

```diff
@@ -64,4 +66,8 @@ export class AgreementComponent {
       }
     }, error => this._notificationService.error('T_ERROR_AGREEMENT'));
   }
+
+  public goBack(): void {
+    this._location.back();
```

Comment:
If the page was opened from a blank browser tab or from another resource a user will be redirected back to that tab or resource.
It is better to redirect user to the root of routing in such case to keep him in the system.

Reply:
> If the page was opened from a blank browser tab
In that case just nothing would be happend and user will be stay on `/agreement` page, because browser history will empty.

But generally your point is valid.